### PR TITLE
In-memory Engine: align config names with spec

### PR DESCRIPTION
**Pull request by overvenus**
### What is changed and how it works?

Issue Number: ref #16141 

What's Changed:

```commit-message
* Fix IME online config change not taking effect.
* Rename `enabled` to `enable`.
* Rename `gc_interval` to `gc_run_interval`.
* Rename `hard_limit_threshold` to `capacity`.
* Rename `soft_limit_threshold` to `evict_threshold`.
* Rename `stop_load_limit_threshold` to `stop_load_threshold`.
```

### Check List

Tests 

- Unit test
- Integration test

### Release note

```release-note
None
```

### CODE REVIEW

File: components/in_memory_engine/src/engine.rs [hunk 1]
```diff
@@ -690,12 +690,12 @@ pub mod tests {
...
-                enabled: true,
-                gc_interval: Default::default(),
+                enable: true,
+                gc_run_interval: Default::default(),
                 load_evict_interval: Default::default(),
-                stop_load_limit_threshold: Some(ReadableSize(300)),
-                soft_limit_threshold: Some(ReadableSize(300)),
-                hard_limit_threshold: Some(ReadableSize(500)),
+                stop_load_threshold: Some(ReadableSize(300)),
+                evict_threshold: Some(ReadableSize(300)),
```
Comment (v01dstar) [inline components/in_memory_engine/src/engine.rs:697]: 👍 This is a much better name.

**Comment by zhangjinpeng87** [comment]
Please make "https://github.com/tikv/tikv/blob/master/etc/config-template.toml" consistent with this change.

**Review by SpadeA-Tang — APPROVED** [review]

File: components/in_memory_engine/src/config.rs [hunk 2]
```diff
@@ -1,20 +1,160 @@
...
+    /// When memory usage reaches this amount, we stop loading regions.
+    // TODO(SpadeA): ultimately we only expose one memory limit to user.
+    // When memory usage reaches this amount, no further load will be
+    // performed.
+    pub stop_load_threshold: Option<ReadableSize>,
+    /// Determines the oldest timestamp (approximately, now - gc_run_interval)
+    /// of the read request the in memory engine can serve.
+    pub gc_run_interval: ReadableDuration,
+    pub load_evict_interval: ReadableDuration,
+    pub expected_region_size: Option<ReadableSize>,
```
Comment (glorv) [inline components/in_memory_engine/src/config.rs:38]: This field is used to determine the region size, thus why not directly use `raftstore.coprocessor.region_split_size` instead? /cc @SpadeA-Tang

**Review by zhangjinpeng87 — APPROVED** [review]
LGTM

**Pull request merged**